### DISCUSSION TITLE
[@types/react] FocusEvent should not narrow type of "target" property

### DISCUSSION THREAD

**Discussion by qtow** [DefinitelyTyped/DefinitelyTyped#65509] [Issues with a @types package]
React's focus and blur events both bubble, so the `target` property is often a descendant of the element the event handler is attached to. Anyone who expects the `target` property to be `EventTarget & Target` _probably_ wants the `currentTarget` property instead, which already exists and is properly typed.

`FocusEvent` inherits from `SyntheticEvent`, which inherits from `BaseSyntheticEvent`, which already has a properly typed `target` property. However, `FocusEvent` [narrows the type of this property](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/7ce9fe81aeb226b0fda1c43495df95d6ce340513/types/react/index.d.ts#L1265). It should avoid explicitly defining the property altogether since it already inherits the correct type.

See also: [this same issue for `SyntheticEvent.target`](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/11508#issuecomment-256045682)

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment]
Yep makes sense. Feel free to file a PR. Don't forget to update the types for older versions (in v15/, v16/ and v17/) and the 5.0 fork. 

We have a hint in other Event interfaces like "if you think this should be..." so when we remove the narrowed target we should c&p that hint there as well.
